### PR TITLE
Enable Kubernetes API for DCS.

### PR DESCRIPTION
**Pull request by Antiarchitect**
Add ability to use Kubernetes directly for DCS purposes. This will avoid using etcd or zookeeper.

### CODE REVIEW

**Comment by Antiarchitect** [comment]
Wait for RBAC fix.

**Comment by Antiarchitect** [comment]
To use K8s built-in DCS abilities you should have this in your config:
```
  etcd:
    enable: false
    deployChart: false
  kubernetes:
    dcs:
      enable: true
```

**Comment by Antiarchitect** [comment]
@linki @paulczar Is there any news on this one?

**Comment by Antiarchitect** [comment]
It works actually on our stage. Not sure if the "app" label fits well for Patroni clusters separation (in my case - pretty well). Maybe we need this thing configurable.

File: incubator/patroni/values.yaml [hunk 1]
```diff
@@ -17,6 +17,11 @@ credentials:
 
 # Distribution Configuration stores
 # Please note that only one of the following stores should be enabled.
+kubernetes:
```
Comment (ese) [inline incubator/patroni/values.yaml:20]: IMHO make sense do kubernetes DCS enable by default and save launch a etcd cluster

Comment (Antiarchitect) [reply incubator/patroni/values.yaml:20]: Will do!

**Comment by Antiarchitect** [comment]
@ese Any updates?

**Comment by ese** [comment]
for me it's ok :+1:

**Comment by Antiarchitect** [comment]
@linki @paulczar Any improvements I should make?

**Comment by laevilgenius** [comment]
Any updates on this?

**Comment by Antiarchitect** [comment]
@linki @paulczar

**Comment by paulczar** [comment]
/retest

need to get these tests passing before we can merge it in.

**Comment by Antiarchitect** [comment]
Could you help me with how to resolve
```
Warning  FailedScheduling   3m45s (x26 over 10m)    default-scheduler   pod has unbound PersistentVolumeClaims (repeated 3 times)
```
on e2e testing

**Comment by ofaz** [comment]
Hey all, 

How has this progressed? Very eager to start using this for some of our projects! If there's anything we can do to help please let me know!

**Comment by paulczar** [comment]
There was a bug on the volume claim template that was stopping it from being bound to the statefulset, I've fixed that and pushed the fix up to your branch.

/lgtm

**Pull request merged**